services/project-files-provider-base.ts

```diff
@@ -0,0 +1,47 @@
+///<reference path="../.d.ts"/>
+"use strict";
+
+import * as path from "path";
+import * as util from "util";
+
+export abstract class ProjectFilesProviderBase implements IProjectFilesProvider {
+	abstract isFileExcluded(filePath: string): boolean;
+	abstract mapFilePath(filePath: string, platform: string): string;
+
+	constructor(private $mobileHelper: Mobile.IMobileHelper,
+		private $options: ICommonOptions) {}
+
+	public getPreparedFilePath(filePath: string): string {
+		let projectFileInfo = this.getProjectFileInfo(filePath);
+		return path.join(path.dirname(filePath), projectFileInfo.onDeviceFileName);
+	}
+
+	public getProjectFileInfo(filePath: string, platform?: string): IProjectFileInfo {
+		let parsed = this.parseFile(filePath, this.$mobileHelper.platformNames, platform || "");
+		if (!parsed) {
+			parsed = this.parseFile(filePath, ["debug", "release"], this.$options.release ? "release" : "debug");
+		}
+
+		return parsed || {
+			filePath: filePath,
+			onDeviceFileName: path.basename(filePath),
+			shouldIncludeFile: true,
+			value: platform
```

Comment:
Consider removing this `value` property - it seems that it it's not used anywhere